conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -113,6 +113,7 @@ def _context(self):
             libs = self._get_libs(cpp_info, pkg_name, pkg_folder, pkg_folder_var)
             self._add_root_lib_target(libs, pkg_name, cpp_info)
         exes = self._get_exes(cpp_info, pkg_name, pkg_folder, pkg_folder_var)
+        sources = self._get_sources(cpp_info, pkg_name, pkg_folder, pkg_folder_var)
```

Comment:
Have you tried to integrate it in the ``self._get_libs()``, and the ``self._get_cmake_lib()`` code? It seems that it would be simpler, as it doesn't need to repeat all the components logic and iteration.